packages/x-charts/src/internals/plugins/featurePlugins/useChartCartesianAxis/useChartCartesianAxis.ts

```diff
@@ -123,6 +196,7 @@ export const useChartCartesianAxis: ChartPlugin<UseChartCartesianAxisSignature<a
     instance,
     params.disableAxisListener,
     isInteractionEnabled,
+    onAxisInteraction,
```

Comment:
Should we wrap this in an `useEventCallback`? 

If `onAxisInteraction` is a function that's created on every render, we might get into infinite loops because we run this `useEffect` whenever `onAxisInteraction` changes, which might call `onAxisInteraction` with `newAxisPointerInteraction` which is a new array every time. If `onAxisInteraction` causes a re-render (e.g., by calling `setState`), then we'll get an infinite loop.

Reply:
We could, but I'm not a big fan of memorizing user functions.

About the inifit loop, I don't think it's an issue since `onAxisInteraction` is only called inside pointer events callback